### PR TITLE
Update/10.0.1

### PR DESCRIPTION
**Pull request by ELundby45**
# Arrow-cpp v10.0.1

upstream: https://github.com/apache/arrow/tree/apache-arrow-10.0.1
release notes: https://arrow.apache.org/release/10.0.1.html

## Actions
- Remove ignore_run_exports
- Remove old unneeded patches 
- Apply changes needed for the removal of `ARROW_PYTHON` flag. (see bleow)
- Ensure shared libraries are being built
- Fix ORC functionality on Windows

## Notes
- `ARROW_PYTHON` was deprecated. The files that were generated by this flag previously are now inside of `pyarrow`. 
   - The other flags that were enabled when this flag was toggle have to be manually enabled now 
   - See https://github.com/apache/arrow/pull/14224#discussion_r981399130 and https://issues.apache.org/jira/browse/ARROW-17868
- C++17 compiler is necessary hence the newer compiler and SDK versions.
- Linter error is about not having any python import calls. But all of the python imports are built by `pyarrow`
- 

## Related PRs (WiP)
- https://github.com/AnacondaRecipes/pyarrow-feedstock/pull/10

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -41,40 +31,42 @@ requirements:
...
-    - uriparser
-    - utf8proc
+    - rapidjson 1.1.0
+    - re2 2022.04.01
+    - snappy 1.1.9
+    - thrift-cpp 0.15.0  # [not (osx and arm64)]
+    - thrift-cpp 0.13.0  # [osx and arm64]
+    - uriparser 0.9.3
+    - utf8proc 2.6.1
     - zlib
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:66]: why those does not have pinnings ?

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/bld.bat [hunk 2]
```diff
@@ -29,15 +33,16 @@ cmake -G "Ninja" ^
...
       -DARROW_WITH_BROTLI=ON ^
       -DARROW_WITH_BZ2=ON ^
       -DARROW_WITH_LZ4=ON ^
       -DARROW_WITH_SNAPPY=ON ^
       -DARROW_WITH_ZLIB=ON ^
       -DARROW_WITH_ZSTD=ON ^
+      -DCMAKE_PREFIX_PATH=%PREFIX% ^
       -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
       -DCMAKE_INSTALL_LIBDIR="%LIBRARY_PREFIX%" ^
       -DPYTHON_EXECUTABLE="%PYTHON%" ^
```
[the remark this answers is not in the document]
Comment (ELundby45) [reply recipe/bld.bat:51]: Enabling GCS requires we [add a new feedstock](https://github.com/conda-forge/google-cloud-cpp-feedstock/blob/main/recipe/meta.yaml) Do you think I should still pursue this?

[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/bld.bat:51]: Let's skip GCS for now then.

**Review by cbouss — APPROVED** [review]

**Pull request merged**